### PR TITLE
Feature add indexed recursive inscriptions endpoint

### LINKED ISSUE #3923: Recursion: Get latest child from parent inscription
Currently you need to loop over all pages to get the latest child. 

To save multiple calls, would it be possible to adopt something similar to sat number, using `at/-1` to get latest...

`/r/children/<INSCRIPTION_ID>/inscriptions/at/-1`

### PR DESCRIPTION
**Pull request by twosatsmaxi**
Fixes #3923 

- [x] Added tests
- [x] Lint

### CODE REVIEW

**Comment by twosatsmaxi** [comment]
@casey @raphjaph can someone review it?

File: src/subcommand/server.rs [hunk 1]
```diff
@@ -232,6 +232,10 @@ impl Server {
           "/r/children/:inscription_id/inscriptions",
           get(Self::child_inscriptions_recursive),
         )
+        .route(
+          "/r/children/:inscription_id/inscriptions/at/:index",
```
Comment (elocremarc) [inline src/subcommand/server.rs:236]: We dont need `/inscriptions/` in the route it is too verbose and is plural. We are only returning a single Id here.
We should follow the format of the first child endpoint not the second one. I never got around to doing this feature on the original child endpoint I was going to but ran out of time.
```diff
-        "/r/children/:inscription_id/inscriptions/at/:index",
+        "/r/children/:inscription_id//at/:index",
```

File: src/subcommand/server.rs [hunk 2]
```diff
@@ -6206,6 +6245,56 @@ next
...
+    });
+
+    server.mine_blocks(1);
+    let latest_child_inscription_id = InscriptionId { txid, index: 69 };
+    let child_json = server.get_json::<api::ChildInscriptionRecursive>(format!(
+      "/r/children/{parent_inscription_id}/inscriptions/at/-1"
+    ));
+    assert_eq!(child_json.id, latest_child_inscription_id);
+  }
+
```
Comment (elocremarc) [inline src/subcommand/server.rs:6297]: Need test for `/at/0` which would be the first child inscription. As well as `/at/1` which is the second child. This checks the off by one error we had in the original sat endpoint PR see discussion about that [here](https://github.com/ordinals/ord/pull/2452#issuecomment-1791815670)

Look at the sat endpoint for reference of `/at/n` tests:

**Review by elocremarc — CHANGES_REQUESTED** [review]

**Comment by elocremarc** [comment]
Upon further review I don't think there is any benefit of limiting ourself to using the Struct `ChildInscriptionRecursive` this is from the newer child endpoint which is a more optmized child endpoint that uses less onchain data. This struct was made for tons of pagination at the cost of data for speed. Considering we are only returning the info of a single inscription we can afford to return everything about that inscription and use the `InscriptionRecursive` struct from `/r/inscription`

**Comment by elocremarc** [comment]
> Upon further review I don't think there is any benefit of limiting ourself to using the Struct ChildInscriptionRecursive

On second thought I think we should only return the ID. 
@casey @raphjaph do you agree? 
This is redundant data from `/r/inscription`.  Its useful on `/r/children/:inscription_id/inscriptions` because it returns 100 inscriptions with data. However this is only returning info about one inscription.

**Comment by twosatsmaxi** [comment]
@raphjaph can you have a look once?

**Comment by twosatsmaxi** [comment]
@casey @raphjaph: still waiting for the review.

File: src/subcommand/server.rs (hunk 1 above)
Comment (twosatsmaxi) [reply src/subcommand/server.rs:236]: sounds good.

File: src/subcommand/server.rs (hunk 2 above)
Comment (twosatsmaxi) [reply src/subcommand/server.rs:6299]: will do.

**Comment by wagedu** [comment]
> > Upon further review I don't think there is any benefit of limiting ourself to using the Struct ChildInscriptionRecursive
> 
> On second thought I think we should only return the ID. @casey @raphjaph do you agree? This is redundant data from `/r/inscription`. Its useful on `/r/children/:inscription_id/inscriptions` because it returns 100 inscriptions with data. However this is only returning info about one inscription.

I'm a bit confused with the lack of hype for `address` info in general. Considering it's the 1st thing listed right after the `ID`, on any inscription page...
I'd vote for the whole data in the 100 children. 
My current case as example:  4 inscriptions with 100 children each. But instead of 4 queries, it necessitates 400 queries, to compare addresses.

**Review by Deviltep — COMMENTED** [review]
```

**Comment by twosatsmaxi** [comment]
@raphjaph @casey can you have a look once and also what do you think about comments by @elocremarc? would like your inputs too before going ahead make those changes, makes sense too me.
The PR has been open since more than a month, would appreicate your reviews.

File: src/subcommand/server.rs (hunk 1 above)
Comment (twosatsmaxi) [reply src/subcommand/server.rs:253]: done.

File: src/subcommand/server.rs (hunk 2 above)
Comment (twosatsmaxi) [reply src/subcommand/server.rs:6533]: done.

**Pull request closed without merging**